rmgpy/data/thermo.py

```diff
@@ -1745,6 +1745,7 @@ def __addGroupThermoData(self, thermoData, database, molecule, atom):
                     data = entry.data
                     comment = entry.label
                     break
+            else: raise DatabaseError("Node {0} points to a non-existant group called {1} in database: {2}".format(node.label, data, database.label))
```

Comment:
To my understanding the while loop above should only execute once, so shouldn't it be an if statement instead?

Reply:
Also there are two ways for the while loop above it to terminate:  
1) node = None 
or
2 ) node.data has something that isn't None
in case 1 it will error when it sets data = node.data

so to reach this while loop we have to start with data is not None

since the while loop only executes once we don't need to check that data is not None in the if statement